streams/src/main/java/org/apache/kafka/streams/processor/assignment/assignors/StickyTaskAssignor.java

```diff
@@ -331,7 +328,7 @@ public Set<ProcessId> findClientsWithoutAssignedTask(final TaskId taskId) {
 
         public double clientLoad(final ProcessId processId) {
             final int capacity = clients.get(processId).numProcessingThreads();
-            final double totalTaskCount = newAssignments.getOrDefault(processId, new HashSet<>()).size();
+            final double totalTaskCount = newAssignments.getOrDefault(processId, KafkaStreamsAssignment.of(processId, new HashSet<>())).tasks().size();
```

Comment:
just a thought: since we require that all ProcessIds present in the input (ie the ApplicationState#getKafkaStreamsState map) are also present in the output assignment, we can just initialize the `newAssignments` field to a map with keyset equal to the #getKafkaStreamsState keyset, and initialized to only an empty map of tasks. That way we don't need this `getOrDefault` at all and should be able to always be safe calling get on any ProcessId -- does that make sense?

Reply:
Yep, I'm already doing the initialization as you suggested, but never got rid of the `getOrDefault`.